crates/nu-engine/src/filesystem/filesystem_shell.rs

```diff
@@ -34,6 +38,8 @@ pub struct FilesystemShell {
     pub(crate) path: String,
     pub(crate) last_path: String,
     pub(crate) mode: FilesystemShellMode,
+    #[cfg(windows)]
+    pub(crate) windows_drives_previous_cwd: Arc<Mutex<IndexMap<String, String>>>,
```

Comment:
Every fs_shell has one instance of windows_drives_previous_cwd. This instance is shared with all other fs_shells and comes from shell_manager.